docs(rfc): deep links structure convention (#3878)
tari-project/tari: RFC/src/RFC-0154_DeepLinksConvencion.md

THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.

## Language

The keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", 
"NOT RECOMMENDED", "MAY" and "OPTIONAL" in this document are to be interpreted as described in 
[BCP 14](https://tools.ietf.org/html/bcp14) (covering RFC2119 and RFC8174) when, and only when, they appear in all capitals, as 
shown here.

## Disclaimer

This document and its content are intended for information purposes only and may be subject to change or update
without notice.

This document may include preliminary concepts that may or may not be in the process of being developed by the Tari
community. The release of this document is intended solely for review and discussion by the community regarding the
technological merits of the potential system outlined herein.

## Goals

The aim of this Request for Comment (RFC) is to specify the deep links structure used in the Tari Aurora project.
The primary motivation is to create a simple, human-readable, and scalable way to structure deep links used by the Tari Aurora clients.

## Description

> Deep links are the URIs with hierarchical components sequence. We can use this sequence to pass and handle data in a standardized and predictable way. To do that, we need to pass three components to the target client:

Reviewer: ```suggestion
Deep links are the URIs with hierarchical components sequence. We can use this sequence to pass and handle data in a standardized and predictable way. To do that, we need to pass three components to the target client: the scheme, command, and data.
```

### Scheme

> The scheme is used to address the client, which will handle the data from the command and data component. In the Tari Aurora project, we're using `tari` scheme to open the wallet app and handle provided data.

Reviewer: ```suggestion
The scheme is used to address the client, which will handle the command and data components. In the Tari Aurora project, we're using the `tari` scheme to open the wallet app and execute the command.
```

### Command

> The command is path string used to pass information about the action that should be performed by the client. The handler can use this info to determine which parser/decoder should be used to deserialize data and what the app suppose to do with the data afterwards.

Reviewer: ```suggestion
The command is a path string used to pass information about the action that should be performed by the client. The handler uses this command to determine how to deserialize the data, before passing it to the command function.
```

> To support multiple domains/networks the first path component should contain a name of corresponded domain/network. The app before handle the data should first check that active account is pointed to the right domain/network and show the error is there a missmatch. 

Reviewer: ```suggestion
To support multiple networks the first path component should be the name of corresponding network. Before proceeding with the command, the client should first check that the active account is pointed to the correct network, or show an error if there is a mismatch. 
```

For simple actions, the command can be defined as a single phrase. For more complex actions, the command should be defined as a multi-path where the second path component should be the name of the action group.

#### Examples:

Simple Actions:
```ignore
mainnet/user_profile
testnet/login
```
Complex Actions:
```ignore
mainnet/payments/send
testnet/payments/request
```

### Data

> The data component is an optional string of key-value pairs used by the parser/decoder to deserialize the data, which the client passes to the command function. The data component should be formatted in the same way as a URL query. The sub-component should have a `?` prefix, the key-value pairs should be separated by `&`, and every key should be separated from the value by `=` character.

Reviewer 2: Feel free to use the specification language defined in IETF RFC8174. 
"Data component MUST be formatted ..." etc.

> The data component is an optional string of key-value pairs used by the parser/decoder to deserialize the data, which the client will later use to handle the action. The data component should be formatted in the same way as a URL query. The sub-component should have a `?` prefix, the key-value pairs should be separated by `&`, and every key should be separated from the value by `=` character.

Reviewer: ```suggestion
The data component is an optional string of key-value pairs used by the parser/decoder to deserialize the data, which the client passes to the command function. The data component should be formatted in the same way as a URL query. The sub-component should have a `?` prefix, the key-value pairs should be separated by `&`, and every key should be separated from the value by `=` character.
```
